src/libsyntax/parse/parser.rs

```diff
@@ -2700,6 +2697,99 @@ impl<'a> Parser<'a> {
         self.maybe_recover_from_bad_qpath(expr, true)
     }
 
+    fn parse_async_macro_or_stmt(
+        &mut self,
+        lo: Span,
+        await_sp: Span,
+    ) -> PResult<'a, (Span, ExprKind)> {
+        Ok(match self.token {
+            token::Not => {
+                // Handle correct `await!(<expr>)`
```

Comment:
Nit: can you move the comment onto the arm itself? Might more easily convey that this is the purpose of the whole arm instead of a comment re. a specific statement in the arm body.